File: frontend/src/api/news.ts
import { apiClient } from '@/api/client'

export interface NewsArticle {
  id: number
  source: string
  category: string
  title: string
  summary: string | null
  url: string
  published_at: string
  crawled_at: string
  heat_score: number
}

export interface NewsListResponse {
  items: NewsArticle[]
  total: number
}

export interface NewsSourceResult {
  source: string
  success: boolean
  fetched_count: number
  error: string | null
}

export interface NewsRefreshResponse {
  success: boolean
  fetched_count: number
  inserted_count: number
  refreshed_at: string
  sources: NewsSourceResult[]
}

export async function fetchNews(
  limit = 50,
  sources?: string[],
  offset = 0
): Promise<NewsListResponse> {
  const { data } = await apiClient.get<NewsListResponse>('/news', {
    params: { limit, offset, ...(sources ? { sources: sources.join(',') } : {}) },
  })
  return data
}

export async function fetchNewsSources(): Promise<string[]> {
  const { data } = await apiClient.get<string[]>('/news/sources')
  return data
}

export async function refreshNews(sources?: string[]): Promise<NewsRefreshResponse> {
  const { data } = await apiClient.post<NewsRefreshResponse>('/news/refresh', { sources })
  return data
}

File: backend/app/schemas/news.py
"""新闻 API 数据结构。"""

from datetime import datetime

from pydantic import BaseModel, Field


class NewsArticleResponse(BaseModel):
    id: int
    source: str
    category: str
    title: str
    summary: str | None
    url: str
    published_at: datetime
    crawled_at: datetime
    heat_score: int = Field(ge=0, le=100)

    model_config = {"from_attributes": True}


class NewsListResponse(BaseModel):
    items: list[NewsArticleResponse]
    total: int


class NewsSourceResult(BaseModel):
    source: str
    success: bool
    fetched_count: int = 0
    error: str | None = None


class NewsRefreshRequest(BaseModel):
    sources: list[str] | None = None


class NewsRefreshResponse(BaseModel):
    success: bool
    fetched_count: int
    inserted_count: int
    refreshed_at: datetime
    sources: list[NewsSourceResult] = Field(default_factory=list)
